File: tutorials/testing/java/elevation_snippet.java
import java.util.List;
import java.util.Random;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.ElevationData;

// a program fragment to access the Elevation data
public class elevation_snippet {
	public static void main(String[] args) throws Exception {

		//create the Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

		// Get a List of Elevation data  records from Bridges
		DataSource ds = bridges.getDataSource();
		ElevationData elev_data = ds.getElevationData(41.33133177632377,
				-98.02593749997456, 42.508577297430456, -96.94531249997696, 0.02);


		System.out.println("Width: " + elev_data.getCols());
		System.out.println("Height: " + elev_data.getRows());
		System.out.println("Resolution: " + elev_data.getCellSize());
		System.out.println("Lower Left Corner: " + elev_data.getxll() + ", " + elev_data.getyll());

		System.out.println("first 10 elevations:");
		for (int i = 0; i < 10; ++i) {
			System.out.println(elev_data.getData()[0][i]);
		}


	}
}

File: tutorials/testing/c++/elevation_snippet.cpp
#include "Bridges.h"
#include "DataSource.h"
#include <iostream>
#include <string>
#include "data_src/ElevationData.h"

using namespace std;
using namespace bridges;

int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	//create the Bridges object, set credentials

	DataSource ds;
	dataset::ElevationData elev_data;

	// Get a List of Elevation data  records from Bridges
	try {
		elev_data = ds.getElevationData(41.03133177632377, -98.02593749997456, 
					42.508577297430456, -96.94531249997696);
	}
	catch (std::string s) {
		std::cerr << "Exception: " << s << "\n";
	}
	catch (char const* s) {
		std::cerr << "Exception: " << s << "\n";
	}

	
	// print the data dimensions  and the firs 10 elevation data values
	cout << "\tWidth: " << elev_data.getCols() << endl
		<< "\tHeight: " << elev_data.getRows() << endl
		<< "\tCell Size: " << elev_data.getCellSize() << endl
		<< "\tLower Left Corner: "  << elev_data.getxll()  << 
				 ", " <<  elev_data.getyll() <<endl;

	cout << "The first 10 elevation values.." << endl;
	for (int k = 0; k < 10; k++)
		cout << elev_data.getVal(0, k) << "  " << endl;
	cout << endl;

	return 0;
}
